README.md

```diff
@@ -1,3 +1,122 @@
+
+This library exposes a unified, portable API for run-time reflection in Scala.
+It currently supports Scala/JVM and Scala.js, but will eventually support Scala Native as well (currently blocked on [scala-native#1279](https://github.com/scala-native/scala-native/issues/1279)).
+To be portable, only the subset of reflection capabilities that is implementable across all platforms is exposed.
+
+## Setup
+
+Add the following line to your (cross-)project's settings in `build.sbt`:
+
+```scala
+libraryDependencies += "org.portable-scala" %%% "reflect" % "0.1.0-SNAPSHOT"
+```
+
+`reflect` 0.1.0-SNAPSHOT supports:
+
+* Scala 2.10.x, 2.11.x, 2.12.x and 2.13.0-M4
+* Scala/JVM
+* Scala.js 0.6.x and 1.0.0-M5
+
+## Usage
+
+### Instantiate a class given its name
+
+In order to reflectively instantiate a class, `reflect` demands that you "enable reflective instantiation" for it.
+This is the case if:
+
+* The class is annotated with `@org.portablescala.reflect.annotation.EnableReflectiveInstantiation`, or
+* The class directly or indirectly extend a class or trait annotated with that annotation.
+
+For example:
+
+```scala
+import org.portablescala.reflect.annotation.EnableReflectiveInstantiation
+
+@EnableReflectiveInstantiation
+class A // discoverable
+
+@EnableReflectiveInstantiation
+trait SuperTrait
+
+class B extends SuperTrait // discoverable
+
+class C extends B // discoverable
+
+class D // NOT discoverable
+```
+
+In addition, a class must satisfy the following properties to be discoverable:
+
+* It must be concrete
+* It must have at least one public constructor
+* It must not be a local class, i.e., defined inside a method
+
+If a class is discoverable, you can use the method `lookupInstantiatableClass` of `org.portablescala.reflect.Reflect` to get an `InstantiatableClass` representing it using:
+
+```scala
+import org.portablescala.reflect._
+
+val cls = Reflect.lookupInstantiatableClass("fully.qualified.ClassName", someClassLoader)
+```
+
+The `someClassLoader` argument is optional, and defaults to the current class loader at call site.
+It is only meaningful on the JVM.
+
+Once you have an `InstantiatableClass`, you can use its methods to instantiate the class:
+
+* `InstantiatableClass`
+  * `val runtimeClass: Class[_]`: the `java.lang.Class[_]` representing the class
+  * `val declaredConstructors: List[InvokableConstructor]`: a list of all invokable constructors of the class
+  * `def newInstance(): Any`: invokes the no-argument constructor of the class to create an instance
+  * `def getConstructor(parameterTypes: Class[_]*): Option[InvokableConstructor]`: find an invokable constructor by the types of its paramters
+* `InvokableConstructor`
+  * `val parameterTypes: List[Class[_]]`: the types of the parameters of that constructor
+  * `newInstance(args: Any*): Any`: invokes the constructor with the given actual arguments (which must conform to the parameter types)
```

Comment:
I would replace all this with an example. The Scaladoc is better suited for listing existing methods (and will not get out of date).